src/rt/cpp/cown.h

```diff
@@ -0,0 +1,165 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+#include <functional>
+#include <tuple>
+#include <utility>
+#include <verona.h>
+
+class cown_ptr_trait
+{
+private:
+  cown_ptr_trait() {}
+
+  template<typename T>
+  friend class cown_ptr;
+};
+
+/**
+ * Smart pointer to represent shared access to a cown.
+ * Can only be used asychronously with `when` to get
+ * underlying access.
+ */
+template<typename T>
+class cown_ptr : cown_ptr_trait
+{
+private:
+  /**
+   * Internal Verona runtime cown for this type.
+   */
+  class ActualCown : public VCown<ActualCown>
+  {
+  private:
+    T value;
+
+  public:
+    ActualCown(T&& t) : value(std::forward<T>(t)) {}
+
+    template<typename TT>
+    friend class acquired_cown;
+  };
+
+  /**
+   * Internal Verona runtime cown for this type.
+   */
+  ActualCown* allocated_cown = nullptr;
+
+  /**
+   * Accesses the internal Verona runtime cown for this handle.
+   */
+  Cown* underlying_cown()
+  {
+    return allocated_cown;
+  }
+
+  /**
+   * Construct a new cown ptr object, actually allocates a runtime cown.
+   *
+   * This is internal, and the `make_cown` below is the public interface,
+   * which has better behaviour for implicit template arguments.
+   */
+  cown_ptr(T&& t) : allocated_cown(new ActualCown(std::forward<T>(t))) {}

```

Comment:
Same question as above.  This looks as if it's move-constructing (or possible copy-constructing?) a cown from an existing `T`, but the use of `std::forward` means that this looks as if it should actually be emplacing a `T`.

I *think* that the `std::forward` here should be `std::move`.